### PR TITLE
[chore] use type in apache receiver

### PR DESCRIPTION
**Pull request by atoulme**
This has the effect of changing the scope instrumentation library name.

### CODE REVIEW

File: receiver/apachereceiver/testdata/scraper/expected.yaml
Comment (dmitryax) [inline receiver/apachereceiver/testdata/scraper/expected.yaml:218]: This is interesting. Is this the way we agreed to go? Maybe we should update the template to `otelcol/{{ .Class }}/{{ .Type }}` so we get `otelcol/receiver/apache`. Or `io.opentelemetry.collector.receiver.apache`, which is closer to examples defined in https://github.com/open-telemetry/opentelemetry-specification/blob/f9c26af89dcb2857772e2b75fb1318a521aa05c8/specification/glossary.md#instrumentation-library

cc @open-telemetry/collector-contrib-approvers

Comment (atoulme) [reply receiver/apachereceiver/testdata/scraper/expected.yaml:218]: Yes, this is an unfortunate side-effect of using type more rigorously. Happy to chat about the best way to manage that, we can indeed also change the template.

Comment (atoulme) [reply receiver/apachereceiver/testdata/scraper/expected.yaml:218]: Please look at https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4849307450/jobs/8641252822?pr=21275 as well for context. Right now the type field in metadata is what we are going to attempt to use to check if a component is registered in otelcontribcol components. If the type does not match the typeStr constant folks declare in factory.go, then the test breaks.

Comment (evan-bradley) [reply receiver/apachereceiver/testdata/scraper/expected.yaml:218]: In terms of the format, the Go SDK uses the module name, which I think would make sense: https://github.com/open-telemetry/opentelemetry-go-contrib/blob/8538a773eb81e44260cb04118ae3002d18abfe32/instrumentation/net/http/otelhttp/config.go#L30. That would make it `github.com/open-telemetry/opentelemetry-collector-contrib/receiver/apachereceiver`.

The [specification examples](https://github.com/open-telemetry/opentelemetry-specification/blob/f9c26af89dcb2857772e2b75fb1318a521aa05c8/specification/overview.md#instrumentation-libraries) for instrumentation libraries use language-specific module formats too, so I think this would make sense.

I'm also fine with the names that @dmitryax suggested, which has the benefit of not being tied to the path of the component in the repository.

File: receiver/apachereceiver/testdata/integration/expected.yaml
Comment (codeboten) [inline receiver/apachereceiver/testdata/integration/expected.yaml:218]: i wouldn't expect this to change unless we agree to it, this is a breaking change for users

**Review by codeboten — DISMISSED** [review]
Marking request for change to ensure an agreement is reached on whether the scope name should be changed.

File: receiver/apachereceiver/testdata/integration/expected.yaml
Comment (atoulme) [reply receiver/apachereceiver/testdata/integration/expected.yaml:218]: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21382 is filed

File: receiver/apachereceiver/testdata/scraper/expected.yaml
Comment (dmitryax) [reply receiver/apachereceiver/testdata/scraper/expected.yaml:218]: @evan-bradley That Makes sense. Let's leave `io.opentelemetry.collector.receiver.apache` for java. I wish we had the shortcut to contrib modules as well to have a  `go.opentelemetry.io/collector-contrib/receiver/apachereceiver`

Comment (Aneurysm9) [reply receiver/apachereceiver/testdata/scraper/expected.yaml:218]: +1 to using the package import path for instrumentation scope name.  I believe it is the best option for Go packages.

Regarding the `go.opentelemetry.io` metadata proxy, it takes data from [here](https://github.com/open-telemetry/opentelemetry-go-vanityurls/blob/main/go.opentelemetry.io/vanity.yaml) and adding an entry for `collector-contrib` should be easy.  I believe there are issues with hosting and management that are being worked through ATM, but I would expect the more significant impediment to its use by collector contrib to simply be the accumulated mass of code already built working with modules at the existing import paths that would need to change.

Comment (dmitryax) [reply receiver/apachereceiver/testdata/scraper/expected.yaml:218]: I created an issue https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21469. @Aneurysm9 can you please mention/link the hosting and management issues there?

**Comment by fatsheep9146** [comment]
I think on hold can be removed since #21382 is closed?

**Comment by atoulme** [comment]
Sure, and we need to rebase and regenerate those files, should be doable.

**Review by djaglowski — APPROVED** [review]

**Review by fatsheep9146 — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

**Pull request merged**
